### PR TITLE
[HINTS] Get point from x hint

### PR DESCRIPTION
**Pull request by Juan-M-V**
# get_point_from_x hint

## Description

Must be merged after #333.
Implemets secp/signature -> get_point_from_x hints

## Checklist
- [x] Unit tests added
- [x] Integration tests added.

### CODE REVIEW

File: src/math_utils.rs [hunk 1]
```diff
@@ -69,10 +89,10 @@ fn igcdex(num_a: BigInt, num_b: BigInt) -> (BigInt, BigInt, BigInt) {
     }
 }
 ///Finds a nonnegative integer x < p such that (m * x) % p == n.
-fn div_mod(n: BigInt, m: BigInt, p: BigInt) -> BigInt {
-    let (a, _, c) = igcdex(m, p.clone());
-    assert_eq!(c, bigint!(1));
-    (n * a).mod_floor(&p)
+pub fn div_mod(n: &BigInt, m: &BigInt, p: &BigInt) -> BigInt {
+    let (a, _, c) = igcdex(m.clone(), p.clone());
```
Comment (Oppen) [inline src/math_utils.rs:93]: No need for clones, let's just use refs.

File: src/math_utils.rs [hunk 2]
```diff
@@ -82,17 +102,25 @@ pub fn ec_add(
...
 /// Assumes the points are given in affine form (x, y) and have different x coordinates.
-pub fn line_slope(point_a: (BigInt, BigInt), point_b: (BigInt, BigInt), prime: &BigInt) -> BigInt {
-    assert!((point_a.0.clone() - point_b.0.clone()) % prime != bigint!(0));
-    div_mod(point_a.1 - point_b.1, point_a.0 - point_b.0, prime.clone())
+pub fn line_slope(
+    point_a: &(BigInt, BigInt),
+    point_b: &(BigInt, BigInt),
+    prime: &BigInt,
+) -> BigInt {
+    assert!((point_a.0.clone() - point_b.0.clone()) % prime != bigint!(0_i32));
```
Comment (Oppen) [inline src/math_utils.rs:118]: Why the clones? Also, let's us `is_zero`.

File: src/math_utils.rs [hunk 3]
```diff
@@ -108,11 +136,11 @@ pub fn ec_double(point: (BigInt, BigInt), alpha: &BigInt, prime: &BigInt) -> (Bi
...
 /// Assumes the point is given in affine form (x, y) and has y != 0.
 pub fn ec_double_slope(point: (BigInt, BigInt), alpha: &BigInt, prime: &BigInt) -> BigInt {
-    assert!(point.1.clone() % prime != bigint!(0));
+    assert!(point.1.clone() % prime != bigint!(0_i32));
     div_mod(
-        bigint!(3) * point.0.clone() * point.0.clone() + alpha,
-        bigint!(2) * point.1,
-        prime.clone(),
+        &(bigint!(3_i32) * point.0.clone() * point.0.clone() + alpha),
+        &(bigint!(2_i32) * point.1),
```
Comment (Oppen) [inline src/math_utils.rs:142]: Avoid clones and use `is_zero` for checks.

File: src/vm/hints/secp/field_utils.rs [hunk 4]
```diff
@@ -39,12 +38,7 @@ pub fn verify_zero(
 
     let pack = pack(val_d0, val_d1, val_d2, &vm.prime);
 
-    //SECP_P = 2**256 - 2**32 - 2**9 - 2**8 - 2**7 - 2**6 - 2**4 - 1
-    let sec_p = bigint_str!(
-        b"115792089237316195423570985008687907853269984665640564039457584007908834671663"
-    );
-
-    let (q, r) = pack.div_rem(&sec_p);
+    let (q, r) = pack.div_rem(&*SECP_P);
```
Comment (Oppen) [inline src/vm/hints/secp/field_utils.rs:41]: Let's try `SECP_P` directly.

**Review by Oppen — CHANGES_REQUESTED** [review]

File: src/vm/hints/secp/field_utils.rs (hunk 4 above)
Comment (Juan-M-V) [reply src/vm/hints/secp/field_utils.rs]: resolved: https://github.com/lambdaclass/cleopatra_cairo/pull/338/commits/a0008d67fc3cf60bc34e3cd8d19773e833d52772

File: src/math_utils.rs (hunk 3 above)
Comment (Juan-M-V) [reply src/math_utils.rs]: resolved:
- https://github.com/lambdaclass/cleopatra_cairo/pull/338/commits/a0008d67fc3cf60bc34e3cd8d19773e833d52772
- https://github.com/lambdaclass/cleopatra_cairo/pull/338/commits/83bde4c8487d60e10b3bd34bc4e9b7cb23857f5c

File: src/math_utils.rs (hunk 2 above)
Comment (Juan-M-V) [reply src/math_utils.rs]: resolved:
- https://github.com/lambdaclass/cleopatra_cairo/commit/a0008d67fc3cf60bc34e3cd8d19773e833d52772
- https://github.com/lambdaclass/cleopatra_cairo/commit/83bde4c8487d60e10b3bd34bc4e9b7cb23857f5c

File: src/math_utils.rs (hunk 1 above)
Comment (Juan-M-V) [reply src/math_utils.rs]: resolved! :D
https://github.com/lambdaclass/cleopatra_cairo/pull/338/commits/0d96379ce5e302e39d889883238fbdbb97e48486

**Review by Oppen — APPROVED** [review]

**Pull request merged**